### PR TITLE
Improve printing of topology info

### PR DESCRIPTION
**Pull request by drroe**
This PR enhances the capabilities of the topology print routines, i.e. `atoms`, `resinfo`, `molinfo`, `bonds`, `angles`, `dihedrals`, etc.
- Output can now be directed to a file via `out <file>`.
- Selection of bonds/angles/dihedrals can now be restricted by additional masks (1 more for bonds, 2 for angles, 3 for dihedrals). For example, if you only want dihedrals containing the atom types N3, CT, CT, and H1 in that order, type `dihedralinfo @%N3 @%CT @%CT @%H1`.
- Formatting has been cleaned up; headers now appear/are hidden as is appropriate for the topology, fields have more sensible widths, and everything is whitespace-delimited.
- Values of bonds/angles/dihedrals will also be printed if a reference structure is specified, e.g.
```
parm myparm.parm7
reference myref.rst7
bonds @%CT reference
```
- Atom types with names larger than 2 characters will now be correctly displayed.
- Adds a test for everything

This was inspired by a feature request from @dacase

### CODE REVIEW

**Comment by hainm** [comment]
@drroe 

should `bonds` be `bondinfo` so it will be similar to `dihedralinfos`?

**Comment by drroe** [comment]
bondinfo is already a command and is an alias for bonds (same thing for
other info commands).

**Comment by hainm** [comment]
thanks. Just realize that I need update `xxxinfo` for pytraj too.

**Comment by drroe** [comment]
If you ever want to know which keywords are associated with a certain functionality check the `Command::Init()` function in `Command.cpp`.

File: src/Exec_Top.cpp [hunk 1]
```diff
@@ -19,105 +20,150 @@ Exec::RetType Exec_ParmInfo::Execute(CpptrajState& State, ArgList& argIn) {
...
+  if (CommonSetup(info, State, argIn, "Bond info")) return CpptrajState::ERR;
+  std::string mask1 = argIn.GetMaskNext();
+  if (info.PrintBondInfo( mask1, argIn.GetMaskNext() )) return CpptrajState::ERR;
   return CpptrajState::OK;
 }
 // -----------------------------------------------------------------------------
 void Exec_AngleInfo::Help() const {
-  mprintf("\t[%s] [<mask>]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print angle info for atoms in <mask> for specified topology (first by default).\n");
+  mprintf("\t[%s] [<mask1>] [<mask2> <mask3>]\n\t[out <file>]\n", DataSetList::TopIdxArgs);
```
Comment (swails) [inline src/Exec_Top.cpp]: In ParmEd, mask2 and mask3 are independently optional.  If you specify 2 masks, the middle atom needs to be in mask2 and either of the other atoms can be in mask1.

It's a little more logic checking, but still useful sometimes.

File: src/Exec_Top.cpp [hunk 2]
```diff
@@ -19,105 +20,150 @@ Exec::RetType Exec_ParmInfo::Execute(CpptrajState& State, ArgList& argIn) {
...
+  std::string mask2 = argIn.GetMaskNext();
+  if (info.PrintAngleInfo( mask1, mask2, argIn.GetMaskNext() )) return CpptrajState::ERR;
   return CpptrajState::OK;
 }
 // -----------------------------------------------------------------------------
 void Exec_DihedralInfo::Help() const {
-  mprintf("\t[%s] [<mask>] [and]\n", DataSetList::TopIdxArgs);
-  mprintf("  Print dihedral info for atoms in <mask> for specified topology (first by default).\n"
-          "  If 'and' is specified dihedral must include all atoms specfied by <mask>.\n");
+  mprintf("\t[%s] [<mask1>] [<mask2> <mask3> <mask4>]\n\t[out <file>]\n", DataSetList::TopIdxArgs);
```
Comment (swails) [inline src/Exec_Top.cpp]: Same comment as above -- Dihedrals can specify either 2, 3, or 4 masks.

Basically what this amounts to is having the defaults for every mask be `*`

**Pull request merged**